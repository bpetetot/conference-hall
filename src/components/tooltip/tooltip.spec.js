/* eslint-disable react/jsx-filename-extension */
/* eslint-env jest */
import React from 'react'
import snap from 'tests/snapshot'

import Tooltip from './tooltip'

jest.mock(
  'popper.js',
  () =>
    class {
      update = jest.fn(() => undefined)

      destroy = jest.fn(() => undefined)
    },
)

const snapshot = (props) =>
  snap(Tooltip)({ tooltip: 'my content', children: <div>Wrapped</div>, ...props })

describe('tooltip', () => {
  it('should set tooltip content', snapshot({}))
  it('should display tooltip inline', snapshot({ inline: true }))
  it('should set placement "auto"', snapshot({ placement: 'auto' }))
  it('should set placement "top"', snapshot({ placement: 'top' }))
  it('should set placement "bottom"', snapshot({ placement: 'bottom' }))
  it('should set placement "left"', snapshot({ placement: 'left' }))
  it('should set placement "right"', snapshot({ placement: 'right' }))
})
