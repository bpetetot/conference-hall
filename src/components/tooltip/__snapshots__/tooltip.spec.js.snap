// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tooltip should display tooltip inline 1`] = `
<button
  aria-label="More info"
  className="cc-tooltip-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "display": "inline",
    }
  }
  type="button"
>
  <div>
    Wrapped
  </div>
  <div
    className="cc-tooltip"
    data-placement=""
    role="tooltip"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="cc-tooltip-arrow"
      style={Object {}}
    />
    <div
      className="cc-tooltip-inner"
    >
      my content
    </div>
  </div>
</button>
`;

exports[`tooltip should set placement "auto" 1`] = `
<button
  aria-label="More info"
  className="cc-tooltip-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "display": "block",
    }
  }
  type="button"
>
  <div>
    Wrapped
  </div>
  <div
    className="cc-tooltip"
    data-placement=""
    role="tooltip"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="cc-tooltip-arrow"
      style={Object {}}
    />
    <div
      className="cc-tooltip-inner"
    >
      my content
    </div>
  </div>
</button>
`;

exports[`tooltip should set placement "bottom" 1`] = `
<button
  aria-label="More info"
  className="cc-tooltip-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "display": "block",
    }
  }
  type="button"
>
  <div>
    Wrapped
  </div>
  <div
    className="cc-tooltip"
    data-placement=""
    role="tooltip"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="cc-tooltip-arrow"
      style={Object {}}
    />
    <div
      className="cc-tooltip-inner"
    >
      my content
    </div>
  </div>
</button>
`;

exports[`tooltip should set placement "left" 1`] = `
<button
  aria-label="More info"
  className="cc-tooltip-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "display": "block",
    }
  }
  type="button"
>
  <div>
    Wrapped
  </div>
  <div
    className="cc-tooltip"
    data-placement=""
    role="tooltip"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="cc-tooltip-arrow"
      style={Object {}}
    />
    <div
      className="cc-tooltip-inner"
    >
      my content
    </div>
  </div>
</button>
`;

exports[`tooltip should set placement "right" 1`] = `
<button
  aria-label="More info"
  className="cc-tooltip-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "display": "block",
    }
  }
  type="button"
>
  <div>
    Wrapped
  </div>
  <div
    className="cc-tooltip"
    data-placement=""
    role="tooltip"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="cc-tooltip-arrow"
      style={Object {}}
    />
    <div
      className="cc-tooltip-inner"
    >
      my content
    </div>
  </div>
</button>
`;

exports[`tooltip should set placement "top" 1`] = `
<button
  aria-label="More info"
  className="cc-tooltip-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "display": "block",
    }
  }
  type="button"
>
  <div>
    Wrapped
  </div>
  <div
    className="cc-tooltip"
    data-placement=""
    role="tooltip"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="cc-tooltip-arrow"
      style={Object {}}
    />
    <div
      className="cc-tooltip-inner"
    >
      my content
    </div>
  </div>
</button>
`;

exports[`tooltip should set tooltip content 1`] = `
<button
  aria-label="More info"
  className="cc-tooltip-button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "display": "block",
    }
  }
  type="button"
>
  <div>
    Wrapped
  </div>
  <div
    className="cc-tooltip"
    data-placement=""
    role="tooltip"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="cc-tooltip-arrow"
      style={Object {}}
    />
    <div
      className="cc-tooltip-inner"
    >
      my content
    </div>
  </div>
</button>
`;
